### PR TITLE
Speed up some over parameterized tests

### PR DESCRIPTION
**Pull request by dabacon**

### CODE REVIEW

File: cirq-core/cirq/ops/clifford_gate_test.py [hunk 1]
```diff
@@ -58,9 +58,12 @@ def _all_rotation_pairs():
         yield (px, flip_x), (pz, flip_z)
 
 
+@functools.cache
 def _all_clifford_gates():
```
Comment (vtomole) [inline cirq-core/cirq/ops/clifford_gate_test.py:62]: Line not touched but might as well add a return type annotation while we're at it.

**Review by vtomole — APPROVED** [review]
LGTM % nit

**Comment by vtomole** [comment]
Looks like `functools.cache` just got added in [Python 3.9](https://docs.python.org/3.9/library/functools.html). Maybe we want to replace it with [lru_cache()](https://docs.python.org/3.9/library/functools.html#functools.lru_cache) for now?

File: cirq-core/cirq/ops/clifford_gate_test.py (hunk 1 above)
Comment (dabacon) [reply cirq-core/cirq/ops/clifford_gate_test.py]: done

Comment (dabacon) [reply cirq-core/cirq/ops/clifford_gate_test.py]: and it made me realize this likely doesn't work since List is not hashable....so changed to tuple!

**Comment by vtomole** [comment]
Naive python question: Why do these functions have different return types?

```python
from typing import Generator, Tuple

def tuple_of_ints() -> Tuple[int, ...]:
    return tuple(
       i for i in range(2)
    )
    
def generator_of_tuple_of_ints() -> Generator[int, None, None]:
    return (
       i for i in range(2)
    )
```

**Comment by dabacon** [comment]
You might think `(x for x in range(2))` is a tuple, but it's not.  It's a generator expression.  So that's why the second one is typed as you have it.   Here is a good ref https://anandology.com/python-practice-book/iterators.html

File: cirq-core/cirq/ops/clifford_gate_test.py [hunk 2]
```diff
@@ -739,7 +740,7 @@ def test_multi_clifford_decompose_by_unitary():
     # Construct a random clifford gate:
     n, num_ops = 5, 20  # because we relied on unitary cannot test large-scale qubits
     gate_candidate = [cirq.X, cirq.Y, cirq.Z, cirq.H, cirq.S, cirq.CNOT, cirq.CZ]
-    for seed in range(100):
+    for seed in range(10):
```
Comment (pavoljuhas) [inline cirq-core/cirq/ops/clifford_gate_test.py:743]: I'd suggest to drop seeding here and replace it with np.random calls and a simple repetition.
The default numpy random generator is getting reproducibly seeded
in [check/pytest](https://github.com/pavoljuhas/Cirq/blob/d55a99a20b996b4afc3cc23822879c026fe3b37d/check/pytest#L44-L45).  Over time there should be more randomness in the generated choices than with the 10 seeds here.

**Review by pavoljuhas — APPROVED** [review]

**Comment by pavoljuhas** [comment]
LGTM, but I have noticed some strange random seeding in the test with seeds from `range(0, 10)` or so.
I think this should be replaced with np.random calls as the default numpy generator is seeded 
from the last commit time; overall the tests would get more randomness with a standard np.random calls.

Perhaps a matter for separate PR.

File: cirq-core/cirq/ops/clifford_gate_test.py (hunk 2 above)
Comment (dabacon) [reply cirq-core/cirq/ops/clifford_gate_test.py:743]: Good call. Done

Comment (pavoljuhas) [reply cirq-core/cirq/ops/clifford_gate_test.py]: Nice!

**Pull request merged**